engine/src/main/java/org/operaton/bpm/engine/impl/util/BitMaskUtil.java

```diff
@@ -99,7 +99,7 @@ public static int getMaskForBit(int bitNumber) {
   }
 
   private static void ensureBitRange(final int bitNumber) {
```

Comment:
Please do me a favor and add a unit test for BitMaskUtil. The fix is valid, but I think the functionality of this util class is critical enough to require a test.

It would be enough for this change to just add positive & negative test for method `ensureBitRange`. It is currently private. Either test it via one of the public methods. Or just remove "private" to make it package protected. It is totally valid to make private methods package-protected for the sake of covering the method by a unit test in the same package.

I'll check the mentioned Sonar issues now. Thanks for the link.